persistence/src/main/java/de/cyface/persistence/PersistenceLayer.java

```diff
@@ -490,55 +494,126 @@ public void deletePoint3dData(final long measurementIdentifier) {
     }
 
     /**
-     * Loads the track of {@link GeoLocation} objects for the provided {@link Measurement}.
-     * TODO [STAD-6]: Slice the tracks into sub tracks when the measurement was paused and resumed.
-     * Right now the full unsliced track is always the only sub track that is returned.
+     * Loads the {@link Track}s for the provided {@link Measurement}.
      * <p>
-     * This method loads the complete track into memory. For large tracks this could slow down the device or even reach
-     * the applications memory limit.
+     * This method loads the complete {@code Track}s into memory. For large {@code Track}s this could slow down the
+     * device or even reach the applications memory limit.
+     *
+     * TODO [MOV-554]: provide a custom list implementation that loads only small portions into memory.
      *
      * TODO [CY-4438]: From the current implementations (MeasurementContentProviderClient loader and resolver.query) is
      * the loader the faster solution. However, we should upgrade the database access as Android changed it's API.
      *
-     * TODO [MOV-554]: provide a custom list implementation that loads only small portions into memory.
-     *
      * @param measurementIdentifier The id of the {@code Measurement} to load the track for.
-     * @return The sub tracks associated with the {@code Measurement}. Right now we return the full track as
-     *         the first sub track which contains list of ordered (by timestamp) {@code GeoLocation}s.
-     *         If no {@code GeoLocation}s exists, an empty list is returned.
+     * @return The {@link Track}s associated with the {@code Measurement}. If no {@code GeoLocation}s exists, an empty
+     *         list is returned.
+     * @throws CursorIsNullException when accessing the {@code ContentProvider} failed
      */
     @SuppressWarnings("unused") // Sdk implementing apps (RS) use this api to display the tracks
-    public List<List<GeoLocation>> loadTrack(final long measurementIdentifier) {
+    public List<Track> loadTracks(final long measurementIdentifier) throws CursorIsNullException {
```

Comment:
As I wanted to have a similar API like in iOS and for better readability I added the Track Class. Thus, the API changed anyway for this method so I also renamed loadTrack to loadTracks.